### PR TITLE
update underscore expression docs

### PR DESCRIPTION
**Pull request by amy-chalk**

### CODE REVIEW

File: underscore.mdx [hunk 1]
```diff
@@ -5,133 +5,145 @@ description: Using underscore expressions to define features
...
-    total_mass: float = _.engine.mass + _.astronauts[_.mass].sum()
-
-+   # Filtering on the `height` column
-+   count_tall_astronauts = _.astronauts[_.height > 2.0].count()
-```
+To compute `User.total_spend`, we needed to create a projection of the `User.transactions` DataFrame limited to only the
+`Transaction.total` column so that the `sum` aggregation could work. In contrast, no projection was needed for
+`num_large_transaction`'s `count` aggregation because `count` works on DataFrames with any number of columns.
+
+## All supported operations
```
Comment (amy-chalk) [inline underscore.mdx:104]: I thought about adding docstrings here, but I felt like it would be 90% "repeat the name of the thing as the definition", and that felt goofy

File: underscore.mdx [hunk 2]
```diff
@@ -5,133 +5,146 @@ description: Using underscore expressions to define features
...
+    total: float
+    sales_tax: float
+    subtotal: float = _.total - _.sales_tax
+
++ @features
++ class User:
++    id: int
++    # implicit has-many relationship with Transaction due to `user_id` above
++    transactions: DataFrame[Transaction]
++    large_transactions: DataFrame[Transaction] = _.transactions[_.total > 1000]
```
Comment (Nathan-Fenner) [inline underscore.mdx:64]: This is currently unsupported - although you can _use_ dataframe expressions inside of underscores, the actual output always has to be a scalar (with the exception of the materialized aggregate features which are a special case).

So for example, you can say

```
large_transaction_count: int = _.transactions[_.total > 1000].count()
```

or

```
large_transaction_total: int = _.transactions[_.total, _.total > 1000].sum()
```

Comment (Nathan-Fenner) [reply underscore.mdx:64]: Eventually I would like to be able to lift this restriction but it is not super easy

File: underscore.mdx [hunk 3]
```diff
@@ -5,133 +5,146 @@ description: Using underscore expressions to define features
...
+* Less than: `<`
+* Less than or equal: `<=`
+* Equal: `==`
+* Not equal: `!=`
+* Boolean and: `&`
+* Boolean or: `|`
+
+Be careful to not use Python's `and`, `or`, `not`, or `is` operators. Python does not allow these operators to be
+overridden. They take precedence and will be evaluated with native Python, so they will not work with Chalk's underscore
+expressions.
```
Comment (Nathan-Fenner) [inline underscore.mdx:124]: I would rephrase slightly - in particular, "evaluated with native Python" kind of sounds like it will work but be slow, but actually you'll just get garbage.

I think just being a bit more brief works here:

> Do not use Python's `and`, `or`, `not`, or `is` operators in underscore expressions. Python does not allow these operators to be overridden, so they will not work with Chalk's underscore expressions. 

I will also try to fix chalkpy so that if you _do_ make this mistake, it will throw an exception. This way you'll notice it's wrong during `chalk apply`.

File: underscore.mdx [hunk 4]
```diff
@@ -5,133 +5,146 @@ description: Using underscore expressions to define features
...
+
+### Aggregations
+* `sum`
```
Comment (Nathan-Fenner) [inline underscore.mdx:127]: It could be helpful to specify some of the edge cases around these operators. e.g. something like

- sum
  - `null` values are skipped while computing `.sum()`
  - if the dataframe is empty or all of its values are `null`, then `0` is returned

- min
  - `null` values are skipped while computing `.min()`
  - if the dataframe is empty or all of its values are `null`, then `null` is returned. Ensure that the corresponding feature is marked as optional to handle this case, or use `_.coalesce` to provide a default value when the `.min()` is `null` 


Also, `approx_percentile` takes a parameter e.g. `approx_percentile(0.75)` returns a value which is approximately equal to the 75% percentile of the values in the dataframe

File: underscore.mdx [hunk 5]
```diff
@@ -5,133 +5,152 @@ description: Using underscore expressions to define features
...
+## Overview
 
-Underscore expressions can be used to define and
-resolve features from operations on other features.
+Underscore expressions are used to derive features from operations on other features. In feature definitions, `_`
+represents a reference to the containing feature class and is used to access other features in the same instance. In
+addition to arithmetic operations, underscore expressions can also be used to filter and aggregate data.
 
-Consider the feature class `RocketEngine` defined below:
+Underscore expressions are useful because Chalk statically analyzes and optimizes them, leading to better performance
```
Comment (michaelwiktorek) [inline underscore.mdx:14]: For this paragraph do we want to emphasize the performance benefits of underscores or the convenience of writing them compared to python resolvers? I don't really know or have an opinion because both are valuable to us and to customers. The convenience of using underscores really shows in the examples compared to the verbosity of actual resolver functions.

**Review by michaelwiktorek — APPROVED** [review]
